File: C++/DFCI-Lib/Coord.cpp
//
//  Coord.cpp
//  DFCI-Lib-CPP
//
//  Created by Patrice Trognon on 28/10/2015.
//  Copyright © 2015 Patrice Trognon. All rights reserved.
//

#include "Coord.hpp"

#include "Globals.h"

#include "LatLng.hpp"
#include "LambertII.hpp"

Coord::Coord(LatLng* pLatLng, LambertII* pLambertII) {
    piLatLng = new LatLng(pLatLng->latitude, pLatLng->longitude);
    piLambertII = new LambertII(pLambertII->x, pLambertII->y);
}

Coord::Coord(const Coord& rCoord) {
    piLambertII = new LambertII(*rCoord.piLambertII);
    piLatLng = new LatLng(*rCoord.piLatLng);
}

Coord::~Coord() {
    safeDeletePtr(piLambertII)
    safeDeletePtr(piLambertII)
}


Coord&
Coord::operator=(const Coord& rCoord) {

    if ( this != &rCoord ) {
        safeDeletePtr(piLambertII)
        piLambertII = new LambertII(*rCoord.piLambertII);
        
        safeDeletePtr(piLatLng)
        piLatLng = new LatLng(*rCoord.piLatLng);
    }
    
    return *this;
}

File: ObjectiveC/DFCI-Lib/Coord.m
//
//  Coord.m
//  DFCI-ObjC
//
//  Created by Patrice Trognon on 19/09/2015.
//  Copyright © 2015 Patrice Trognon. All rights reserved.
//

#import "Coord.h"

#import "LatLng.h"
#import "LambertII.h"

@implementation Coord

-(instancetype)initWithLatLng:(LatLng*)latLng lambertII:(LambertII*)lambertII {
    self = [super init];
    if ( self ) {
        self.latLng = latLng ;
        self.lambertII = lambertII ;
    }
    return self ;
}

-(instancetype)initWithLatitude:(double)latitude longitude:(double)longitude XLambert:(double)x YLambert:(double)y {
    self = [super init];
    if ( self ) {
        self.latLng = [[LatLng alloc]initWithLatitude:latitude longitude:longitude];
        self.lambertII = [[LambertII alloc]initWithXLambert:x YLambert:y];
    }
    return self ;
}

-(instancetype)initLatLngFromXLambert:(double)x YLambert:(double)y {
    self = [super init];
    if ( self ) {
        self.lambertII = [[LambertII alloc]initWithXLambert:x YLambert:y];
    }
    return self ;
}

-(instancetype)initLambertFromLatitude:(double)latitude longitude:(double)longitude {
    self = [super init];
    if ( self ) {
        self.latLng = [[LatLng alloc]initWithLatitude:latitude longitude:longitude];
    }
    return self ;
}

@end
